database/src/test/kotlin/no/nav/su/se/bakover/database/TestDataHelper.kt

```diff
@@ -367,107 +398,231 @@ internal class TestDataHelper(
+                        sakId = sakId,
+                        søknadId = søknadId,
+                        journalpostId = journalpostId,
+                        søknadInnhold = søknadInnhold,
+                    ),
+                )
+            }
+        }.let {
+            val medOppgave = it.second.medOppgave(oppgaveId).also { søknad ->
+                søknadRepo.oppdaterOppgaveId(søknad)
+            }
+            Pair(
+                sakRepo.hentSak(it.first.id)
+                    ?: throw IllegalStateException("Fant ikke sak rett etter vi opprettet den."),
+                medOppgave,
+            )
         }
     }
 
-    fun nyOversendtUtbetalingMedKvittering(
-        periode: Periode = stønadsperiode.periode,
+    /**
+     * Persisterer:
+     * 1. [Søknadsbehandling.Iverksatt.Innvilget]
+     * 1. [Utbetaling.OversendtUtbetaling]:
+     *    1. [Utbetaling.OversendtUtbetaling.UtenKvittering]
+     *    1. [Utbetaling.OversendtUtbetaling.MedKvittering]
+     * 1. [VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling]
+     */
+    fun persisterVedtakMedInnvilgetSøknadsbehandlingOgOversendtUtbetalingMedKvittering(
+        sakId: UUID = UUID.randomUUID(),
+        søknadId: UUID = UUID.randomUUID(),
+        stønadsperiode: Stønadsperiode = stønadsperiode2021,
         avstemmingsnøkkel: Avstemmingsnøkkel = no.nav.su.se.bakover.database.avstemmingsnøkkel,
-        utbetalingslinjer: NonEmptyList<Utbetalingslinje> = nonEmptyListOf(utbetalingslinje(periode)),
-    ): Pair<Søknadsbehandling.Iverksatt.Innvilget, Utbetaling.OversendtUtbetaling.MedKvittering> {
-        val utenKvittering = nyIverksattInnvilget(
-            periode = periode,
+        utbetalingslinjer: NonEmptyList<Utbetalingslinje> = nonEmptyListOf(utbetalingslinje(stønadsperiode.periode)),
+        utbetalingId: UUID30 = UUID30.randomUUID(),
+        epsFnr: Fnr? = null,
+        vilkårsvurderinger: Vilkårsvurderinger.Søknadsbehandling = vilkårsvurderingerSøknadsbehandlingInnvilget(
+            stønadsperiode.periode,
+        ),
+        grunnlagsdata: Grunnlagsdata = if (epsFnr != null) grunnlagsdataMedEpsMedFradrag(
+            periode = stønadsperiode.periode,
+            epsFnr = epsFnr,
+        ) else grunnlagsdataEnsligMedFradrag(stønadsperiode.periode),
+        søknadsbehandling: Søknadsbehandling.Iverksatt.Innvilget = persisterSøknadsbehandlingIverksattInnvilget(
+            sakId = sakId,
+            søknadId = søknadId,
+            epsFnr = epsFnr,
+            stønadsperiode = stønadsperiode,
+            vilkårsvurderinger = vilkårsvurderinger,
+            grunnlagsdata = grunnlagsdata,
+        ).second,
+    ): Triple<Sak, VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling, Utbetaling.OversendtUtbetaling.MedKvittering> {
+        assert(søknadsbehandling.sakId == sakId && søknadsbehandling.søknad.sakId == sakId && søknadsbehandling.søknad.id == søknadId)
```

Comment:
Det føles som at det eksisterer en "bro" mellom `test-common` og dette et sted. Kunne det vært en mulighet at `TestDataHelper` ikke sto for selve opprettelsen av objektene, men kun tok ansvar for å lagre dem? F.eks hvis man i testene brukte hjelpefunksjonene i `test-common` for å opprette test-data, for deretter å sende inn hele greia til `TestDataHelper` som gikk igjennom "hele treet" og lagret? Vet ikke om det er mulig eller om vi havner i "lås" noe sted, men det er kanskje verdt å utforske?

Reply:
Nei, det vil ikke fungere, siden vi må sende de eksakte tilstandstypene i riktig rekkefølge til lagring. Så det er i teorien bare det første objektet i chainen man kan hente fra test-common. Dersom vi hadde laget mer generelle upsert funksjoner i repoene ville det gått.